gammapy/modeling/models/spatial.py

```diff
@@ -1289,6 +1290,104 @@ def plot_interactive(self, ax=None, geom=None, **kwargs):
         super().plot_interactive(ax=ax, geom=geom, **kwargs)
 
 
+class TemplateNDSpatialModel(SpatialModel):
+    """A model generated from a ND map where extra dimensions define the parameter space.
+
+    Parameters
+    ----------
+    map : `~gammapy.maps.WcsNDMap`
+        Map template.
+    meta : dict, optional
+        Meta information, meta['filename'] will be used for serialization
+    interp_kwargs : dict
+        Interpolation keyword arguments passed to `gammapy.maps.Map.interp_by_pix`.
+        Default arguments are {'method': 'linear', 'fill_value': 0, "values_scale": "log"}.
+    """
+
+    tag = ["TemplateNDSpatialModel", "templateND"]
+
+    def __init__(self, map, interp_kwargs=None, meta=None, filename=None):
+        self._map = map.copy()
```

Comment:
Shouldn't you check the type of the `Map`? One should be able to evaluate on spatial quantities.